domains/Discussions/Tests/Feature/QuestionsUpdateTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Domains\Discussions\Tests\Feature;
+
+use Domains\Accounts\Database\Factories\UserFactory;
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Database\Factories\QuestionFactory;
+use Domains\Discussions\Models\Question;
+use Faker\Factory;
+use Faker\Generator;
+use Illuminate\Http\Response;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Str;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class QuestionsUpdateTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private Generator $faker;
+    private User $user;
+    private Question $question;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->faker = Factory::create();
+        $this->user = UserFactory::new()->create();
+        $this->question = QuestionFactory::new(['author_id' => $this->user->id])->create();
+    }
+
+    /** @test */
+    public function it_updates_questions(): void
+    {
+        Carbon::setTestNow();
+        $payload = [
+            'title' => $this->faker->title,
+            'description' => $this->faker->paragraph,
+        ];
+        $response = $this->actingAs($this->user)
+            ->call(
+                'PATCH',
+                route('discussions.questions.update', ['questionId' => $this->question->id]),
+                $payload
+            );
+
+        self::assertResponseStatus(Response::HTTP_NO_CONTENT);
```

Comment:
We should be using $this, instead of self.

Reply:
Ok, changed, but then shouldn't we keep it consistent across the project? Take this "Links" tests for example, sometime we use `self` other times we use `$this`...

Reply:
It's just that we shouldn't call a non-static function statically :smiley: 

Some functions we call are declared as static, and so we should use self or static (accordingly) and other are operating under an instance of the class (thus, not static). PHP can be a little funky with this, and all because of backward compatibility. Here's a link to understand this a little bit more. https://thephp.cc/news/2017/07/dont-call-instance-methods-statically